source/NVDAObjects/UIA/VisualStudio.py

```diff
@@ -53,6 +54,22 @@ class IntelliSenseLiveRegion(UIA):
 }
 
 
+class CompletionToolTip(ToolTip):
+	""" A tool tip for which duplicate open events can be fired.
+	"""
+
+	_lastToolTipOpenedInfo = (None, None)  #: Keeps track of the last ToolTipOpened event (text, time)
+
+	def event_UIA_toolTipOpened(self):
+		oldText, oldTime = self._lastToolTipOpenedInfo
+		newText = self.name
+		newTime = time.time()
+		self.__class__._lastToolTipOpenedInfo = (newText, newTime)
+		if newText == oldText and oldTime is not None and (newTime - oldTime) < 0.2:
```

Comment:
I think this really needs a comment to explain why this is done.

For instance, it is probably worth pointing out that a user may intentionally be trying to hear the same tooltip twice, hence the 0.2 seconds. I think the value could be larger (eg 1 second). Given that the text must be equal consecutively, we should look to approach the limit of how rapidly a user could ever intentionally try to trigger the same tooltip twice in a row.